doc/manual/src/glossary.md

```diff
@@ -3,17 +3,24 @@
 - [derivation]{#gloss-derivation}
 
   A description of a build task. The result of a derivation is a
-  store object. Derivations are typically specified in Nix expressions
+  store object. Derivations created by Nix expressions are specified
   using the [`derivation` primitive](./language/derivations.md). These are
   translated into low-level *store derivations* (implicitly by
-  `nix-env` and `nix-build`, or explicitly by `nix-instantiate`).
+  `nix-build`, or explicitly by `nix-instantiate`).
+
+  Package definitions, such as in [nixpkgs], are written using one or
+  more derivations, typically with the use of helper functions such as `mkDerivation`.
 
   [derivation]: #gloss-derivation
+  [nixpkgs]: https://github.com/NixOS/nixpkgs
```

Comment:
```suggestion

  [derivation]: #gloss-derivation
```

Same argument as in the other comment: This is the Nix manual, the Nixpkgs manual should be capable of explaining the role of `mkDerivation` well enough. We now have nix.dev providing an entry point for beginners, clarifying that Nixpkgs is built on top of Nix, with the Nix language tutorial clearly highlighting that `derivation` is a low-level primitive that you'll never see or use in practice. I understand the argument that backlinks are valuable for discoverability and such, but not in this case where we're still fighting beginner confusion over separation of concerns.

Reply:
> package definitions

Yes -- no. I don't have the issue by @roberth handy, but essentially: What's even a package? There's no such concept to be found here, to the best of my knowledge.

Also, more practically, Nix can be used to produce any file system data, whatever the problem domain this may encode, and I'm trying to capture that in the manual by being agnostic to applications.

Reply:
Well yes, but I'm not suggesting changing the definition of derivation in terms of packages, just to mention that derivations are very often used for making package definitions with Nix. And yes derivations are very agnostic (which is great!), but Nix *is* "the purely functional package manager", and plenty of other parts of the Nix manual reference the concept of "packages". I don't think it'd be overly presumptuous to say that "packages" are, or at least were, the motivation for derivations to exist, even if they can do more than that.

Reply:
It's true packages are mentioned here and there, but that's essentially artifacts that I remove when I work on a section containing them. I'm trying to get rid of terms that instill the wrong ideas or are effectively meaningless in this context, e.g. "installation". Eventually I will suggest rewording the front page of the manual.

Reply:
Yes. Nix is cool and could be used for lots of things. Is it? No. Not by anyone sane. It's fundamentally designed at it's core at the moment to be connected to package management and that needs to be acknowledged and explained in the documentation. If you want to describe that it's more versatile that's fine, but fucking document what it actually does

Reply:
Packages don't just "install software" they're simply a device that copies a set of files to a standardized location and registers them with a database for management — optionally transforming them, such as building software, patches, etc. — even typical package managers can install fonts, config files, other resources. A derivation *is* essentially a package, Nix's value is that it's more dynamic about how packages are ultimately built and installed. But don't pretend it's not making packages.